### PR TITLE
Add custom check

### LINKED ISSUE #1: Add correction of problems by script
For some problems the correction can't be done by a simple diff, sometimes there are more than one possible correct output, or the output is a float point that has to hold some given precision. In cases like this the problem setter should be able to correct a problem by a script

### PR DESCRIPTION
**Pull request by simiaosimis**
This closes https://github.com/BROnlineJudge/broj/issues/1

For now the user have to write a python script that define the `check` method. The method has access to the input, the expected output and the user output as strings and can check however the user wants since it returns true if it should be AC and false otherwise.

### CODE REVIEW

File: ej/models.py [hunk 1]
```diff
@@ -8,6 +8,8 @@
 class Problem(db.Entity):
     title = Required(str, unique=True)
     time_limit = Required(int)
+    custom_check = Required(bool)
```
Comment (CaioIcy) [inline ej/models.py]: @simiaosimis Why do we need the extra **bool** field? I think one nullable **str** field as *custom_check* would be enough.

File: tests/test_judge.py [hunk 2]
```diff
@@ -170,6 +170,8 @@ def do_problem_mock(title, time_limit, inputs, outputs):
     mock_problem = MagicMock()
     mock_problem.title = title
     mock_problem.time_limit = time_limit
+    mock_problem.custom_check = False
```
Comment (CaioIcy) [inline tests/test_judge.py]: Where are the custom script check tests?

**Review by CaioIcy — CHANGES_REQUESTED** [review]
Minor changes + test needed for approval

File: ej/models.py (hunk 1 above)
Comment (simiaosimis) [reply ej/models.py]: I was thinking about a future form to register a new problem where it would have a checkbox `custom check`, thats why.

Comment (CaioIcy) [reply ej/models.py]: When a custom check script is necessary, the field would be filled with the script code. When it is not, the field would be nil. This is why I think we don't need the bool field.

Comment (simiaosimis) [reply ej/models.py]: As the case where you have a script but don't wanna use can be a very specific application, I will make the modifications and keep it simpler

**Review by CaioIcy — APPROVED** [review]

**Pull request merged**